openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,155 @@
-            license_info = get_license_info(license_url=license_url)
-            image_store.add_item(
-                foreign_landing_url=foreign_url,
-                image_url=image_url,
-                license_info=license_info,
-                foreign_identifier=foreign_id,
-                width=width,
-                height=height,
-                title=title,
-                meta_data=metadata,
-                creator=creators,
+            offset = prev_query_params["offset"] + self.batch_limit
+        return {**self._common_args, "offset": offset}
+
+    @staticmethod
+    def _get_data_from_response(response_json) -> dict | None:
+        if response_json and response_json.get("message", "").lower() == "success.":
+            return response_json.get("data")
+        return None
+
+    def get_batch_data(self, response_json) -> list | None:
+        return self._get_data_from_response(response_json)
+
+    @staticmethod
+    def _get_license_url(rights_info):
+        elements = html.fromstring(rights_info.get("description", ""))
+        cc_links = [
+            link
+            for _, _, link, _ in elements.iterlinks()
+            if "https://creativecommons.org/" in link
+        ]
+        license_url = cc_links[0] if len(cc_links) == 1 else None
+        return license_url
+
+    @staticmethod
+    def _get_image_sizes(image):
+        height, width = None, None
+        size_list = image.get("derivatives", "")
+        if isinstance(size_list, list):
+            # Pull out the largest sized image we have access to
+            # https://www.brooklynmuseum.org/api/new-documentation/#definition-Image
+            size_type = image.get("largest_derivative", "")
+            for size in size_list:
+                if size.get("size", "") == size_type:
+                    height = size.get("height")
+                    width = size.get("width")
+        return height, width
+
+    @staticmethod
+    def _get_metadata(data):
+        return {
+            "accession_number": data.get("accession_number"),
+            "date": data.get("object_date"),
+            "description": data.get("description"),
+            "medium": data.get("medium"),
+            "credit_line": data.get("credit_line"),
+            "classification": data.get("classification"),
+        }
+
+    @staticmethod
+    def _get_creators(data):
+        artists_info = data.get("artists")
+        if isinstance(artists_info, list):
+            creators_list = (
+                artists.get("name")
+                for artists in artists_info
+                if artists.get("rank") == 1
             )
+            creator = next(creators_list, None)
+        else:
+            creator = None
+        return creator
+
+    @staticmethod
+    def _handle_object_data(data, license_url) -> list[dict]:
+        images = []
+        image_info = data.get("images")
+        if image_info is not None:
+            id_ = data.get("id", "")
```

Comment:
I think we should `continue` if `id_` is None here. Otherwise, we are getting an invalid `foreign_url` on line 105.